tasks/adjust_config.yml

```diff
@@ -32,41 +32,21 @@
           cmd: |-
             docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ {{ item.appType }} --type={{ item.type }} --value={{ item.value }} -- {{ item.appConfig }} {{ item.configName }}
       with_items:
-        - appType: "config:app:set"
-          appConfig: "previewgenerator"
-          configName: "squareSizes"
-          type: string
-          value: "{{ nextcloud_preview_squareSizes }}"
-        - appType: "config:app:set"
-          appConfig: "previewgenerator"
-          configName: "widthSizes"
-          type: string
-          value: "{{ nextcloud_preview_widthSizes }}"
-        - appType: "config:app:set"
-          appConfig: "previewgenerator"
-          configName: "heightSizes"
-          type: string
-          value: "{{ nextcloud_preview_heightSizes }}"
```

Comment:
When removing or renaming variables in other roles, we don't just drop them silently, but rather create deprecation tasks in `tasks/validate_config.yml`.

We already have some [existing deprecation tasks](https://github.com/mother-of-all-self-hosting/ansible-role-nextcloud/blob/fd4b567e2b02a972a92a5e73ecd8f1b254bc4c47/tasks/validate_config.yml#L17-L25), so adding new ones can be done easily.

Reply:
Good point! Although the kept variables does not impact the execution, yet their presence may cause later trouble or make the user believe they are useful. I think the error should be raised when the variable is defined. However, the only way I know to check is the usage of the `is defined` construct, that is not used in the playbooks (as you mentioned in your previous remark: https://github.com/mother-of-all-self-hosting/ansible-role-nextcloud/pull/20#discussion_r1676302196). Can you point me to something that checks the presence of a variable and also in line with the rest of the playbook?

Reply:
To check for presence, you:

1. Drop it from your role, so that it's not present by default. Any presence that remains must be due to the user's configuration
2.  You then use a `when` check like this: `'nextcloud_preview_widthSizes' in vars`.

In my previous comment, I've linked to [how the Nextcloud role does it](https://github.com/mother-of-all-self-hosting/ansible-role-nextcloud/blob/fd4b567e2b02a972a92a5e73ecd8f1b254bc4c47/tasks/validate_config.yml#L17-L25).

You can either copy that more complicated way or get inspired from `matrix-synapse` (see [here](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/cb7726f4a8b3082e1fdab8949c60f07f5b6bdee7/roles/custom/matrix-synapse/tasks/validate_config.yml#L56-L95)).

Reply:
I think the solution used in daad44d3ceb012c0848ae1f40b4c20c2947b6810 works.